src/module.c

```diff
@@ -7874,6 +7877,42 @@ int RM_ModuleTypeReplaceValue(RedisModuleKey *key, moduleType *mt, void *new_val
     return REDISMODULE_OK;
 }
 
+/* For a specified command, parse its arguments and return an array the
+ * contains the indexes of all keys. This function is essnetially a
+ * more efficient way to do COMMAND GETKEYS.
+ *
+ * A NULL value indicates no keys, or an unknown command.
+ *
+ * NOTE: The returned array is not a Redis Module object so it does not
+ * get automatically freed even when auto-memory is used. The caller
+ * must explicitly call RM_Free() to free it.
+ */
+int *RM_GetCommandKeys(RedisModuleCtx *ctx, const char *cmdname, RedisModuleString **argv, int argc, int *num_keys) {
+    UNUSED(ctx);
+    struct redisCommand *cmd;
+    int *res = NULL;
+
+    if ((cmd = lookupCommandByCString(cmdname)) == NULL)
+        return NULL;
+
+    getKeysResult result = GETKEYS_RESULT_INIT;
+    getKeysFromCommand(cmd, argv, argc, &result);
+
+    *num_keys = result.numkeys;
+    if (!result.numkeys)
+        return NULL;
```

Comment:
i suggest to free the `result` here.
just in case one of the implementation allocated a heap result and eventually returns 0 keys.